File: Dubstep.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;
    
    string result = "";
    int i = 0;
    
    while (i < s.length()) {
        if (s.substr(i, 3) == "WUB") {
            if (result != "" && result[result.length() - 1] != ' ') {
                result += ' ';
            }
            i += 3;
        } else {
            result += s[i];
            i++;
        }
    }
    
    cout << result << endl;
    
    return 0;
}

File: Dubstep.java
import java.util.Scanner;

public class Dubstep {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String s = scanner.nextLine();
        String result = s.replaceAll("WUB", " ").trim().replaceAll(" +", " ");
        
        System.out.println(result);
        
        scanner.close();
    }
}